### PR TITLE
chore: consolidate quote list into route

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Consolidates the conditional rendering of compact version of `TradeQuotes` into a route to simplify state management within `TradeInput` while maintaining the state of quotes and trade input. This move is in preparation for #7255 which requires adding yet another conditionally rendered interface which also requires persisting trade input state, so rather than doubling down on complexity, this PR refactors the trade input to make more general use of routes with persistent trade input state.

There are some benefits with this change:
* Simplifies trade input component which is already massively overcomplicated
* Persists quote fetching state across different trade routes to negate the need to re-fetch when going back to the trade input route. Polling is maintained, so stale data is not a concern.
* Sets a pattern for #7255 where we can implement the flow similar to the quote list and trade confirm UIs
* Fixes janky animation when TradeInput is mounted where the quote list appears and collapses immedately

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

prep for #7255

Moderate risk. Makes fairly substantial changes to the code structure of the trade input components, which if not implemented correctly could result is failing trades or inability to trade.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

The most likely things to break:
* visual layout of the trade input, quote list, etc
* state of the trade input (values, loading state etc) when opening/closing the compact quote list

Test the trade input page, getting quotes, changing input etc for the following:
* Wide desktop view on trade page
    * Side-by-side quote list works as expected
* Narrow desktop view on trade page
    * clicking the quote button to show the compact quote list
    * going back to trade input
* Mobile view on trade page
    * clicking the quote button to show the compact quote list
    * going back to trade input
* Desktop view on asset page
    * clicking the quote button to show the compact quote list
    * going back to trade input

## Screenshots (if applicable)

Desktop asset page:

https://github.com/shapeshift/web/assets/125113430/c8ecbc3f-8d9e-4021-954c-12173d613c81

Desktop wide view on trade page:

https://github.com/shapeshift/web/assets/125113430/27c0db26-29f5-44c7-863e-2140810adaa2

Desktop narrow on trade page:

https://github.com/shapeshift/web/assets/125113430/c0d03f36-a692-499c-8894-814facf4a362

Mobile view on trade page:

https://github.com/shapeshift/web/assets/125113430/a34f145f-fc9e-4378-8a48-aa2fa88e725b

### CODE REVIEW

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 1]
```diff
@@ -242,15 +234,6 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
...
 
-  useEffect(() => {
-    // WARNING: do not remove.
-    // clear the confirmed quote on mount to prevent stale data affecting the selectors
-    dispatch(tradeQuoteSlice.actions.resetConfirmedQuote())
-    // clear the active quote index on mount to prevent stale data affecting the selectors
-    dispatch(tradeQuoteSlice.actions.resetActiveQuote())
-    dispatch(tradeInput.actions.setSlippagePreferencePercentage(undefined))
-  }, [dispatch])
-
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:253]: Note the above removal. All of the routes in the parent component `MultiHopTrade` benefit from persistent store state across mounts, but we DO clear both the `tradeQuoteSlice` and  `tradeInputSlice` completely when `MultiHopTrade` is mounted, so we're safe.

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, conceptually happy with this guy. One blocking issue which seems to be a regression of this PR, otherwise happy to get this guy in 🙏🏽 

- Selected route is persisted ✅ 
- Execution is borked on mobile viewports (❓ ish) 

https://jam.dev/c/b17a8812-c106-4f85-a61b-524bc2653a7a

Note, I wasn't able to *consistently* repro but for context, haven't noticed this one in develop before. Perhaps a me thing and a false alarm, switching between desktop and mobile viewports? See example of a happy subsequent trade:

https://jam.dev/c/894cd5f1-981d-47e7-a683-90d459c4884b

- Execution is still happy on desktop viewport

https://jam.dev/c/f712a6d1-0909-4413-afba-0ae49825c79d

- Noticed initiating a trade then backing out of it before completion and initiating another one routes to completion of the previous one - wasn't able to repro on develop 🚫 

https://jam.dev/c/6f31eae5-95ee-48e8-aa2a-c466be6980b3

- Unrelated to this PR: noticed a flash of content of icons between routes, though it is also present in develop, only more visible now that we do things well and are more graceful in selected quote  persistence

**Comment by NeOMakinG** [comment]
> Tested locally, conceptually happy with this guy. One blocking issue which seems to be a regression of this PR, otherwise happy to get this guy in 🙏🏽
> 
> * Selected route is persisted ✅
> * Execution is borked on mobile viewports (❓ ish)
> 
> https://jam.dev/c/b17a8812-c106-4f85-a61b-524bc2653a7a
> 
> Note, I wasn't able to _consistently_ repro but for context, haven't noticed this one in develop before. Perhaps a me thing and a false alarm, switching between desktop and mobile viewports? See example of a happy subsequent trade:
> 
> https://jam.dev/c/894cd5f1-981d-47e7-a683-90d459c4884b
> 
> * Execution is still happy on desktop viewport
> 
> https://jam.dev/c/f712a6d1-0909-4413-afba-0ae49825c79d
> 
> * Noticed initiating a trade then backing out of it before completion and initiating another one routes to completion of the previous one - wasn't able to repro on develop 🚫
> 
> https://jam.dev/c/6f31eae5-95ee-48e8-aa2a-c466be6980b3
> 
> * Unrelated to this PR: noticed a flash of content of icons between routes, though it is also present in develop, only more visible now that we do things well and are more graceful in selected quote  persistence

I reproduced the same bug on develop, seems unrelated to this PR:

https://jam.dev/c/5d4f55dc-af6f-46d5-a770-d21b55356742

**Review by gomesalexandre — APPROVED** [review]
Stamping after @NeOMakinG confirmation the spotted regression isn't one!

**Comment by NeOMakinG** [comment]
Smallish tests of the swapper to make it more confortable to merge:

https://jam.dev/c/76679a17-8f9b-4b21-9078-00291cda7ee4

**Pull request merged**